src/client/sandbox/cookie/frame-sync.js

```diff
@@ -0,0 +1,107 @@
+import Promise from 'pinkie';
+import INTERNAL_PROPS from '../../../processing/dom/internal-properties';
+import createIntegerIdGenerator from '../../utils/integer-id-generator';
+
+const SYNC_COOKIE_START_EVENT = 'hammerhead|event|sync-cookie-start';
+const SYNC_COOKIE_DONE_EVENT  = 'hammerhead|event|sync-cookie-done';
+
+export default class FrameSync {
+    constructor (win, cookieSandbox, messageSandbox) {
+        this.win            = win;
+        this.cookieSandbox  = cookieSandbox;
+        this.messageSandbox = messageSandbox;
+
+        this.messageIdGenerator = createIntegerIdGenerator();
+
+        this.resolvers = {};
```

Comment:
`resolvers` is a good name for a massive. For the object I suggest something like `resolversMap`...